schematic/store/synapse.py

```diff
@@ -1946,6 +1949,50 @@ def replaceTable(synStore, tableToLoad: pd.DataFrame = None, tableName: str = No
         existing_table.drop(columns = ['ROW_ID', 'ROW_VERSION'], inplace = True)
         return existingTableId
     
+
+    def _get_schematic_db_creds(synStore):
+        username = None
+        authtoken = None
+
+
+        # Get access token from environment variable if available
+        env_access_token = os.getenv("SYNAPSE_ACCESS_TOKEN")
+        if env_access_token:
+            authtoken = env_access_token
+
+        # Get token from authorization header
+        if 'Authorization' in synStore.syn.default_headers:
+            authtoken = synStore.syn.default_headers['Authorization'].split('Bearer ')[-1]
+
+        # retrive credentials from synapse object
+        synapse_object_creds = synStore.syn.credentials
+        if hasattr(synapse_object_creds, 'username'):
+            username = synapse_object_creds.username
+        if hasattr(synapse_object_creds, '_token'):
+            authtoken = synapse_object_creds.secret
+
+        # Try getting creds from .synapseConfig file if it exists
+        if os.path.exists(CONFIG.SYNAPSE_CONFIG_PATH):
+            config = synStore.syn.getConfigFile(CONFIG.SYNAPSE_CONFIG_PATH)
+
+            # check which credentials are provided in file
+            if config.has_option('authentication', 'username'):
+                username = config.get('authentication', 'username')
+            if config.has_option('authentication', 'authtoken'):
+                authtoken = config.get('authentication', 'authtoken')
+        
+        # raise error if required credentials are not found
+        if not username and not authtoken:
+            raise NameError(
+                "Username or authtoken credentials could not be found in the environment, synapse object, or the .synapseConfig file"
```

Comment:
if both "username" and "authtoken" are not available, should the error message say something like "Username **and** authtoken credentials XXX"

Reply:
> if both "username" and "authtoken" are not available, should the error message say something like "Username **and** authtoken credentials XXX"

yes, it should say "and", thanks for catching!


> Can we combine the two raise statements and do something like:
> 
> ```
>         if not username or not authtoken:
>             raise NameError(
>                 "Username or authtoken credentials could not be found in the environment, synapse object, or the .synapseConfig file"
> ```

I wanted to separate out the error messages so that users would know exactly what was missing with less chance for confusion from saying "this or this is missing"